lib/src/command/upgrade.dart

```diff
@@ -44,14 +61,163 @@ class UpgradeCommand extends PubCommand {
       log.warning(log.yellow(
           'The --packages-dir flag is no longer used and does nothing.'));
     }
-    await entrypoint.acquireDependencies(SolveType.UPGRADE,
-        useLatest: argResults.rest,
-        dryRun: argResults['dry-run'],
-        precompile: argResults['precompile']);
+
+    if (argResults.wasParsed('breaking')) {
+      final useLatest = argResults.rest;
+
+      final rootPubspec = entrypoint.root.pubspec;
+      final resolvablePubspec =
```

Comment:
If I invoke: `pub upgrade --breaking analyzer` I would expect only the analyzer package to be upgraded - I don't think this works for that, you need to use `argResults.rest` somehow.

Reply:
This already happens in `useLatest`, but that's a poor choice of parameter name... Perhaps `upgradeOnly` is better. Or loosen constraints is better..

Reply:
Apologies for the bad parameter name - was borrowing the naming from `resolveVersions` without adapting it. Invoking `pub upgrade --breaking analyzer` should upgrade only `analyzer` - I have a test for that in `test/upgrade/breaking/applies_changes_test.dart` - compare the test in lines 12 to 47 versus 112 to 146!

Reply:
Sorry - I was reading this wrong. useLatest is indeed the same as argResults.rest.